python/xoscar/collective/core.py

```diff
@@ -0,0 +1,321 @@
+# Copyright 2022-2023 XProbe Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import hashlib
+import os
+from collections import defaultdict
+from typing import Any, Dict, List, Optional
+
+from .. import Actor, actor_ref
+from ..context import get_context
+from .common import (
+    INVOKE_ERROR_MESSAGE,
+    RANK_ADDRESS_ENV_KEY,
+    AllReduceAlgorithm,
+    CollectiveReduceOp,
+)
+from .process_group import ProcessGroup, ProcessGroupGloo
+from .utils import get_rank_address_via_env
+
+
+class RankActor(Actor):
+    def __init__(
+        self,
+        rank: int,
+        world: int,
+        backend: str = "gloo",
+        pg_options: Optional[ProcessGroup.Options] = None,
+        *args,
+        **kwargs,
+    ):
+        self._rank = rank
+        self._world = world
+        self._backend = backend
+        self.name_to_pg: Dict[str, Dict[str, "ProcessGroup"]] = defaultdict(dict)
+        self._pg_options = pg_options
+
+    @classmethod
+    def default_uid(cls):
+        return "RankActor"
+
+    async def __post_create__(self):
+        os.environ[RANK_ADDRESS_ENV_KEY] = self.address
```

Comment:
> Why set the address to an env and get_rank_address_via_env?

Here aims to reduce a parameter ``address`` on the collective interface. Since we only support call the collective interfaces on the target process, the rank address can be removed in this way. Otherwise we need to provide an extra parameter to the interface.